syntax = "proto3";

package historian.v1;

option go_package = "github.com/ahmet/historian/go-services/pkg/proto/historian/v1";

message SensorData {
  string sensor_id = 1;
  double value = 2;
  int64 timestamp_ms = 3;
  int32 quality = 4; // 0=Bad, 1=Good
}

message LogEntry {
  string level = 1;
  string message = 2;
  int64 timestamp_ms = 3;
  string service = 4;
}

message UserAction {
  string user_id = 1;
  string action = 2;
  string resource = 3;
  int64 timestamp_ms = 4;
}
